backend/fms_core/template_importer/row_handlers/sample_qc/sample.py

```diff
@@ -48,6 +48,10 @@ def process_row_inner(self, sample, sample_information, process_measurement, pro
                                                                               quantity_flag=sample_information['quantity_flag'],
                                                                               quality_flag=sample_information['quality_flag'])
 
+            # Add a warning if the sample has failed qc
+            if any([sample_information['quantity_flag'] == 'Failed', sample_information['quality_flag'] == 'Failed']):
+                self.warnings["flags"] = (f"Source sample {sample_obj.name} has failed QC.")
```

Comment:
I would modify the message here to instead indicate that you are about to overwrite a failed QC flag.